controllers/installation_controller.go

```diff
@@ -724,6 +727,62 @@ func (r *InstallationReconciler) ReconcileRegistry(ctx context.Context, in *v1be
 	return nil
 }
 
+// ReconcileHAStatus reconciles the HA migration status condition for the installation.
+// This status is based on the HA condition being set, the Registry deployment having two running + healthy replicas,
+// and the kotsadm rqlite statefulset having three healthy replicas.
+func (r *InstallationReconciler) ReconcileHAStatus(ctx context.Context, in *v1beta1.Installation) error {
+	if in == nil {
+		return nil
+	}
+
+	if !in.Spec.HighAvailability {
+		in.Status.SetCondition(metav1.Condition{
+			Type:               HAConditionType,
+			Status:             metav1.ConditionFalse,
+			Reason:             "HA Not Enabled",
+			ObservedGeneration: in.Generation,
+		})
+		return nil
+	}
+
+	registryReady, err := k8sutil.GetChartHealth(ctx, r.Client, "registry")
```

Comment:
should check for airgap. would also recommend checking health of seaweedfs chart

Reply:
🤦